### PR TITLE
Updating instructions for using clang format 

### PR DESCRIPTION
**Pull request by matthiasrichter**

### CODE REVIEW

**Comment by sawenzel** [comment]
Fantastic! Thanks.

File: README.md [hunk 1]
```diff
@@ -37,21 +37,47 @@ The code organisation is described [here](doc/CodeOrganization.md).
...
 
 ### Formatting
-##### Using clang-format
+The project uses `clang-format` to push for a common code formatting according
+the the `clang-format` configuration files in this repository. With an adiabatic
+approach, all changes have to follow the formatting rules. A tool script can be
+used to integrate the formatting into `git` and suggest formatting only for
+changed lines.
+
+##### Install `clang-format` git integration
```
Comment (ktf) [inline README.md]: Why not simply including it in O2?

Comment (matthiasrichter) [reply README.md]: true, however not sure if it belongs there

Comment (ktf) [reply README.md]: IMHO, telling people to wget files is not what we want to do. What we support is either "alibuild" or standard "brew install" / "apt-get install". We should avoid proliferation of installation mechanism for different tools.

Comment (ktf) [reply README.md]: @dberzano

Comment (dberzano) [reply README.md]: I fully agree with @ktf. We can possibly have the script distributed with O2 in the utilities folder.

**Pull request merged**